[BUG] MFA fails to install on Python 3.12.3 using pip

**Describe the issue**
Running `pip install Montreal-Forced-Aligner` fails when building `biopython==1.79` under a **Python 3.12.3** virtual environment. It looks like this old version of biopython is not compatible with Python 3.12. I created a fork of the MFA repo and removed the **==1.79** restriction which resulted in a successful installation although I didn't test MFA.

**For Reproducing your issue**
Run `pip install Montreal-Forced-Aligner` in a Python 3.12 environment.

**Log file**
```
      copying Bio/SeqIO/_twoBitIO.c -> build/lib.linux-x86_64-cpython-312/Bio/SeqIO
      copying Bio/PDB/QCPSuperimposer/qcprotmodule.c -> build/lib.linux-x86_64-cpython-312/Bio/PDB/QCPSuperimposer
      running build_ext
      building 'Bio.Align._aligners' extension
      creating build/temp.linux-x86_64-cpython-312/Bio/Align
      x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O2 -Wall -fPIC -I/home/dev/Development/asr/bilmalti-api/aws-api/batch_v2/venv/include -I/usr/include/python3.12 -c Bio/Align/_aligners.c -o build/temp.linux-x86_64-cpython-312/Bio/Align/_aligners.o
      Bio/Align/_aligners.c: In function ‘set_alphabet’:
      Bio/Align/_aligners.c:1786:18: error: ‘PyUnicode_WCHAR_KIND’ undeclared (first use in this function); did you mean ‘PyUnicode_4BYTE_KIND’?
       1786 |             case PyUnicode_WCHAR_KIND:
            |                  ^~~~~~~~~~~~~~~~~~~~
            |                  PyUnicode_4BYTE_KIND
      Bio/Align/_aligners.c:1786:18: note: each undeclared identifier is reported only once for each function it appears in
      Bio/Align/_aligners.c: In function ‘sequence_converter’:
      Bio/Align/_aligners.c:6664:18: error: ‘PyUnicode_WCHAR_KIND’ undeclared (first use in this function); did you mean ‘PyUnicode_4BYTE_KIND’?
       6664 |             case PyUnicode_WCHAR_KIND:
            |                  ^~~~~~~~~~~~~~~~~~~~
            |                  PyUnicode_4BYTE_KIND
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for biopython
Failed to build biopython
```

--- a/ci/docker_environment.yaml
+++ b/ci/docker_environment.yaml
@@ -19,7 +19,7 @@ dependencies:
   - baumwelch
   - ngram
   - praatio=6.0.0
-  - biopython=1.79
+  - biopython
   - sqlalchemy>=2.0
   - git
   - pgvector

--- a/environment.yml
+++ b/environment.yml
@@ -21,7 +21,7 @@ dependencies:
   - baumwelch
   - ngram
   - praatio=6.0.0
-  - biopython=1.79
+  - biopython
   - sqlalchemy>=2.0
   - pgvector
   - pgvector-python

--- a/github_environment.yml
+++ b/github_environment.yml
@@ -18,7 +18,7 @@ dependencies:
   - baumwelch
   - ngram
   - praatio=6.0.0
-  - biopython=1.79
+  - biopython
   - sqlalchemy>=2.0
   - pgvector
   - pgvector-python

--- a/montreal_forced_aligner/helper.py
+++ b/montreal_forced_aligner/helper.py
@@ -12,6 +12,7 @@ import json
 import logging
 import re
 import typing
+import warnings
 from contextlib import contextmanager
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Type

--- a/montreal_forced_aligner/helper.py
+++ b/montreal_forced_aligner/helper.py
@@ -19,10 +19,13 @@ from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Type
 import dataclassy
 import numpy
 import yaml
-from Bio import pairwise2
 from rich.console import Console
 from rich.logging import RichHandler
 from rich.theme import Theme
+
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    from Bio import pairwise2
 
 if TYPE_CHECKING:
     from kalpy.fstext.lexicon import LexiconCompiler

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,7 +6,7 @@ numpy
 scipy
 scikit-learn
 requests
-biopython==1.79
+biopython
 dataclassy
 sqlalchemy>=2.0
 click

--- a/rtd_environment.yml
+++ b/rtd_environment.yml
@@ -17,7 +17,7 @@ dependencies:
   - scikit-learn
   - hdbscan
   - psycopg2
-  - biopython=1.79
+  - biopython
   - click
   - setuptools_scm
   - importlib_metadata

--- a/setup.cfg
+++ b/setup.cfg
@@ -33,7 +33,7 @@ keywords = phonology
 [options]
 packages = find:
 install_requires =
-    biopython<=1.79
+    biopython
     click
     dataclassy
     kneed

